monitoring/cgroup.go

```diff
@@ -152,6 +167,25 @@ func parseProcMounts(content []byte) (mounts []mountPoint, err error) {
 	return mounts, nil
 }
 
+// ReaderWithContext wraps a reader with a context.
+type ReaderWithContext struct {
```

Comment:
Nit: This looks generic (and useful) enough to be in a some kind of "utils" package.